.github/workflows/lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Lint
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+jobs:
+  lint:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14.x"
+      - name: Install dependencies
+        run: npm ci
+      - name: Check prettier
```

Comment:
In astro & docs we don't fail CI for prettier, we just have an action that formats and auto commits the result if needs be. Don't know what you think but it's slightly friendlier.

Reply:
do we have a reusable workflow / github action for that? what perms does it need?

Reply:
Okay, pushed the format action, though I don't know whether it has perms; copied it from prettier-plugin-astro